Makefile

```diff
@@ -69,7 +69,12 @@ ocaml_version:
 ocaml_word_size:
 	@if ! ocamlopt -config | grep "word_size:" | grep $(WORD_SIZE); then echo "invalid machine word size, expected $(WORD_SIZE)" ; exit 1; fi
 
-ocaml_checks: ocaml_version ocaml_word_size
+check_opam_switch:
+ifneq ($(DISABLE_CHECK_OPAM_SWITCH), true)
+	check_opam_switch opam.export
+endif
+
+ocaml_checks: ocaml_version ocaml_word_size check_opam_switch
```

Comment:
one thing I've always wondered, is why aren't we commenting all the rules in the makefile? IMO we should